source.go

```diff
@@ -119,8 +119,12 @@ func (s *Source) Open(ctx context.Context, pos opencdc.Position) error {
 	return nil
 }
 
-func (s *Source) Read(ctx context.Context) (opencdc.Record, error) {
-	return s.iterator.Next(ctx)
+func (s *Source) ReadN(ctx context.Context, n int) ([]opencdc.Record, error) {
+	if n <= 0 {
+		return []opencdc.Record{}, nil
+	}
```

Comment:
I know you're trying to be careful, but I'd just remove this to prevent other developers from thinking it needs to be checked. The SDK should provide the correct `n` that is always at least `1`, otherwise that should be considered a bug in the SDK.

Reply:
@lovromazgon yeah, you're right. I recall seeing that check in the SDK.